lib/helper.js

```diff
@@ -139,16 +139,35 @@ function withContext(Component, context) {
  * this method is not needed if you are using babel
  *
  * @method jsxTranspile
+ * @param {Boolean} instrument the original code
  **/
-function jsxTranspile() {
+function jsxTranspile(instrument) {
     var fs = require('fs');
     var babel = require('babel');
 
-    require.extensions['.jsx'] = function (module, filename) {
-        var content = fs.readFileSync(filename, 'utf8');
-        var compiled = babel.transform(content, {nonStandard: true});
-        return module._compile(compiled.code, filename);
-    };
+    if (instrument !== undefined && instrument && instrument === 'true') {
```

Comment:
@akshayp  i prefer `jenkins-mocha` setting an ENV variable for the coverage. e.g. JENKINS_MOCHA=[true | false].  Next, we can pass the ENV variable to the jsxTranspile().

```javacript
var jsx = require('../../index').jsxTranspile(process.env.JENKIN_MOCHA_COVERAGE);
```

Also, i dump the `process.argv` with jeknin-mocha, and I could not read --no-coverage from argv.  Here's the log.

```bash
[ '/usr/local/bin/node',
  '/Users/stanleyn/Projects/public/git/yahoo/jsx-test/node_modules/jenkins-mocha/node_modules/mocha/bin/_mocha',
  '--reporter',
  '/Users/stanleyn/Projects/public/git/yahoo/jsx-test/node_modules/jenkins-mocha/node_modules/spec-xunit-file',
  '--colors',
  'example/test/CheckboxWithLabel.test.js',
  'example/test/UnnamedComponent.test.js',
  'test/**/*.test.js' ]
[ '/usr/local/bin/node',
  '/Users/stanleyn/Projects/public/git/yahoo/jsx-test/node_modules/jenkins-mocha/node_modules/mocha/bin/_mocha',
  '--reporter',
  '/Users/stanleyn/Projects/public/git/yahoo/jsx-test/node_modules/jenkins-mocha/node_modules/spec-xunit-file',
  '--colors',
  'example/test/CheckboxWithLabel.test.js',
  'example/test/UnnamedComponent.test.js',
  'test/**/*.test.js' ]
```